### PR TITLE
Fix invalid format of generated bom by incomplete package.json files

### LINKED ISSUE #31: `null` entries in dependsOn
Hi,

This behaviour is using `@cyclonedx/webpack-plugin` v1.0.1. Unfortunately I'm having a fair bit of difficulty reproducing it in a minimal example (I'm seeing different behaviour in our closed source app) but I thought it might still be worth reporting.

For the closed source app, I am seeing the following entry in the generated bom.json:

```json5
{
    "ref": "pkg:npm/%40reach/alert@0.16.0",
    "dependsOn": [
        "pkg:npm/%40reach/visually-hidden@0.16.0",
        null
    ]
}
```

In [my attempt at a minimal repro](https://github.com/aidansteele/cyclonedx-webpack-repro), I don't get a null in the `dependsOn` array, but I do get an empty object in the `dependencies` array:

```json5
    {
      "ref": "pkg:npm/%40reach/alert@0.16.0",
      "dependsOn": [
        "pkg:npm/react@17.0.2",
        "pkg:npm/prop-types@15.7.2",
        "pkg:npm/react-dom@17.0.2",
        "pkg:npm/%40reach/visually-hidden@0.16.0"
      ]
    },
    {
      "ref": "pkg:npm/react@17.0.2",
      "dependsOn": [
        "pkg:npm/object-assign@4.1.1"
      ]
    },
    {
      "ref": "pkg:npm/prop-types@15.7.2"
    },
    {
      "ref": "pkg:npm/react-dom@17.0.2",
      "dependsOn": [
        "pkg:npm/scheduler@0.20.2"
      ]
    },
    {
      "ref": "pkg:npm/%40reach/visually-hidden@0.16.0"
    },
    {},
```

I will continue to try to pare down the closed source webpack config (it's big) and see if I can reproduce the nulls in a sample for you. Or let me know if there are specific things I can do to help debug this.

### PR DESCRIPTION
**Pull request by peschuster**
Some packages (like e.g. babel/runtime) contain package.json files in sub directories that don't contain all necessary data for generating boms.

This PR introduces a solution to iterate through parent directories until a package.json with at least a name property is found and prevents incomplete dependency refs from being added to the bom to not generate output that does not comply to the cyclonedx specification.

This should also fix #31.

### CODE REVIEW

**Comment by jkowalleck** [comment]
tried to reproduce the example regarding `@babel/runtime`.
could not find evidences for the described behavior. tried with version 7.18.3 -- did not find any internal/shipped/bundled packages 

in addition to your fix, @peschuster , could you provide an integration test so the behavior can be reproduced and fixed for good?

**Comment by peschuster** [comment]
This here is the package.json that breaks the output in my project (using babel/runtime 7.17.2): https://github.com/babel/babel/blob/main/packages/babel-runtime/helpers/esm/package.json

It is a react project. I will try to create an integration test for it.

**Comment by peschuster** [comment]
@jkowalleck An react example with webpack5 and babel-runtime is now part of the integrations test and the tests are green again (it took some time to find a way through all the pitfalls...)

**Comment by jkowalleck** [comment]
Thank you for the fix.
Could you add an entry to the `HISTORY.md` file?

**Comment by jkowalleck** [comment]
> Thank you for the fix. Could you add an entry to the `HISTORY.md` file?

nevermind, will add it in release preparations

File: tests/integration/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -1481,3 +1481,658 @@ exports[`integration webpack5 with vue2 generated xml file: dist/.bom/bom.xml 1`
   </dependencies>
 </bom>"
 `;
+
+exports[`integration webpack5 with react generated json file: dist/.bom/bom.json 1`] = `
```
[the remark this answers is not in the document]
Comment (peschuster) [reply tests/integration/__snapshots__/index.test.js.snap:1485]: Yes I did and it looks correct to me.

**Pull request merged**